### ISSUE TITLE
Black flash during bitrate switch from 30fps to 60fps

### ISSUE THREAD

**Issue by juechemparathy** [google/ExoPlayer#6362]
### Issue description
Black flash happens on switching from lower bitrate on 30fps to higher bitrate on 60fps.

### Reproduction steps
1. Start playback with good bandwidth and a playlist with 30fps as lowest bitrate and 60fps highest streams. 
2. Wait for few seconds to switch to highest bitrate.
Result: Black screen during the transition during bitrate switch

Link to test content - **Widevine-DRM-DASH-Live**
mpd file -> https://gist.github.com/juechemparathy/c891e565e89e6f12c4a98ac6e17188a4
Stream url -> emailed to dev.exoplayer@gmail.com
Let me know if you guys need more info.Thanks.

### A full bug report captured from the device

From EventLogger

2019-08-29 17:54:45.535 8958-8958/com.yahoo.mobile.client.android.example.yvideosdkdemo D/EventLogger: decoderInputFormatChanged [11.10, 4.10, window=0, video, id=C, mimeType=video/avc, **bitrate=1069178,** codecs=avc1.4d001e, res=640x360, **fps=30.0]**

2019-08-29 17:54:53.910 8958-8958/com.yahoo.mobile.client.android.example.yvideosdkdemo D/EventLogger: decoderInputFormatChanged [19.47, 11.71, window=0, video, id=G, mimeType=video/avc, **bitrate=5949747,** codecs=avc1.640020, res=1280x720, **fps=60.0]`**

 -------   Black screen appears after above log -------

### Version of ExoPlayer being used
2.9.5

### Device(s) and version(s) of Android being used
S8

**Comment by mikulbhatt** [comment]
@juechemparathy can you provide some test stream? also the stream you are trying is Widevine DRM protected DASH right? Do you observe similar behavior on HLS ? can you confirm on that?

**Comment by juechemparathy** [comment]
@mikulbhatt - I don't have an HLS test stream with same setup. Stream url has been emailed to dev.exoplayer@gmail.com

**Comment by juechemparathy** [comment]
@marcbaechinger  - Let me know if you need any help on reproducing the black flash we see during the switch. Thanks.

**Comment by marcbaechinger** [comment]
Hey @juechemparathy Sorry I messed it up:
-> Token expired 353755 seconds ago   

:( Sorry, can you send another url?

**Comment by juechemparathy** [comment]
@marcbaechinger - Please find the test url here - `https://content-vmg.uplynk.com/event/071815f671914422991c2b17dfc91dd6.mpd?tc=1&exp=1567547393&rn=2331&ct=e&cid=071815f671914422991c2b17dfc91dd6&iph=b8b50fcc399d90a482ab2fffe366ebf42bf98a7abf58076d4481fe0a03dcce6b&ad.dnt=true&ad.m.appBundle=com.yahoo.mobile.client.android.example.yvideosdkdemo&ad.m.device=android-app&ad.m.session=H4KCKQM8&ad.r=vsdk-demo-app&ad.rssid=dfef6af5-46c6-3ad2-a93b-57f3a1ff27de&ad.width=962&cdn=ec&expand=nfl&nielsen=true&resolutions=0-720&secure=true&sig=a5e49b9d2a64959e7e8f4cb33b76d022b766e41d59d43ac4ce8a69f40a5bce2e`

**Comment by marcbaechinger** [comment]
Thanks @juechemparathy!

The stream is protected. I found a comment in the mpd telling me what license server url to use but it seems not to work with that. Did you play that stream with some special setup? Can you play it in the demo app by adding it to media.exolist.json?

**Comment by mikulbhatt** [comment]
Hi @marcbaechinger please use following config

```
{  
   "name":"Live - Uplynk DASH Encrypted with UnEncrypted Ads",
   "uri":"https://content-vmg.uplynk.com/event/071815f671914422991c2b17dfc91dd6.mpd?tc=1&exp=1567547393&rn=2331&ct=e&cid=071815f671914422991c2b17dfc91dd6&iph=b8b50fcc399d90a482ab2fffe366ebf42bf98a7abf58076d4481fe0a03dcce6b&ad.dnt=true&ad.m.appBundle=com.yahoo.mobile.client.android.example.yvideosdkdemo&ad.m.device=android-app&ad.m.session=H4KCKQM8&ad.r=vsdk-demo-app&ad.rssid=dfef6af5-46c6-3ad2-a93b-57f3a1ff27de&ad.width=962&cdn=ec&expand=nfl&nielsen=true&resolutions=0-720&secure=true&sig=a5e49b9d2a64959e7e8f4cb33b76d022b766e41d59d43ac4ce8a69f40a5bce2e",
   "drm_scheme":"widevine",
   "drm_multi_session":true,
   "drm_license_url":"https://content.uplynk.com/wv",
   "drm_key_request_properties":{  
      "X-DASH-SEND-ALL-KEYS":"1"
   },
   "extension":"mpd"
}
```

**Comment by juechemparathy** [comment]
> Hi @marcbaechinger please use following config
> 
> ```
> {  
>    "name":"Live - Uplynk DASH Encrypted with UnEncrypted Ads",
>    "uri":"https://content-vmg.uplynk.com/event/071815f671914422991c2b17dfc91dd6.mpd?tc=1&exp=1567547393&rn=2331&ct=e&cid=071815f671914422991c2b17dfc91dd6&iph=b8b50fcc399d90a482ab2fffe366ebf42bf98a7abf58076d4481fe0a03dcce6b&ad.dnt=true&ad.m.appBundle=com.yahoo.mobile.client.android.example.yvideosdkdemo&ad.m.device=android-app&ad.m.session=H4KCKQM8&ad.r=vsdk-demo-app&ad.rssid=dfef6af5-46c6-3ad2-a93b-57f3a1ff27de&ad.width=962&cdn=ec&expand=nfl&nielsen=true&resolutions=0-720&secure=true&sig=a5e49b9d2a64959e7e8f4cb33b76d022b766e41d59d43ac4ce8a69f40a5bce2e",
>    "drm_scheme":"widevine",
>    "drm_multi_session":true,
>    "drm_license_url":"https://content.uplynk.com/wv",
>    "drm_key_request_properties":{  
>       "X-DASH-SEND-ALL-KEYS":"1"
>    },
>    "extension":"mpd"
> }
> ```

@marcbaechinger  - Seems the issue is being addressed with 2.10.4 with the decoder updates. The above test stream will be taken down by end of this week(09/21/2019 ). Thanks.
